pkgs/by-name/ot/otb/package.nix

```diff
@@ -0,0 +1,113 @@
+{
+  cmake,
+  fetchFromGitHub,
+  makeWrapper,
+  lib,
+  stdenv,
+  swig,
+  which,
+  boost,
+  curl,
+  gdal,
+  itk_4_13,
+  libsvm,
+  libgeotiff,
+  muparser,
+  muparserx,
+  opencv,
+  perl,
+  python3,
+  extraPythonPackages ? ps: with ps; [],
+  shark,
+  tinyxml,
+}:
+
+let
+    pythonInputs = with python3.pkgs; [
+      numpy
+    ] ++ (extraPythonPackages python3.pkgs);
+in
+  stdenv.mkDerivation (finalAttrs: {
+    pname = "otb";
+    version = "9.0.0";
+
+    src = fetchFromGitHub {
+      owner = "orfeotoolbox";
+      repo = "otb";
+      rev = "${finalAttrs.version}";
+      hash = "sha256-Ut2aimQL6Reg62iceoaM7/nRuEV8PBWtOK7KFHKp0ws=";
+    };
+
+    nativeBuildInputs = [
+      cmake
+      makeWrapper
+      swig
+      which
+    ];
+
+    buildInputs = [
+      boost
+      curl
+      gdal
+      itk_4_13
+      libsvm
+      libgeotiff
+      muparser
+      muparserx
+      opencv
+      perl
+      python3
+      shark
+      tinyxml
+    ] ++ pythonInputs;
+
+
+    # https://www.orfeo-toolbox.org/CookBook/CompilingOTBFromSource.html#native-build-with-system-dependencies
+    # activates all modules and python by default
+    cmakeFlags = [
+      "-DOTB_WRAP_PYTHON=ON"
+      "-DOTB_BUILD_FeaturesExtraction=ON"
+      "-DOTB_BUILD_Hyperspectral=ON"
+      "-DOTB_BUILD_Learning=ON"
+      "-DOTB_BUILD_Miscellaneous=ON"
+      "-DOTB_BUILD_RemoteModules=ON"
+      "-DOTB_BUILD_SAR=ON"
+      "-DOTB_BUILD_Segmentation=ON"
+      "-DOTB_BUILD_StereoProcessing=ON"
+      "-DBUILD_TESTING=OFF"
+    ];
+
+
+    propagatedBuildInputs = [
```

Comment:
Why do we want to propagate all of this?

Reply:
Thanks @SuperSandro2000.

As OTB allows to build (remote) modules (officially not part of) on top of it, so we propagate the inputs to able to build remote modules based on OTB via nix as well. As I wasn't sure if all are needed so I tried to follow similar to ITK [c.f](https://github.com/NixOS/nixpkgs/blob/a046c1202e11b62cbede5385ba64908feb7bfac4/pkgs/development/libraries/itk/generic.nix#L95)

Example of remote modules for OTB: 
- https://github.com/remicres/otbtf
- https://github.com/remicres/otb-prefetch
- https://gitlab.orfeo-toolbox.org/orfeotoolbox/otb/-/tree/develop/Modules/Remote?ref_type=heads

Reply:
Not sure how this will integrate with nix. First of all propagatedBuildInputs are merged into buildInputs, so we don't need anything in buildInputs that is already in propagatedBuildInputs.

The other things, we don't want to propagate things if possible. Are those modules build with nix? If so we should move things to passthru. If not, then we just keep this ugly workaround for now until we have something better.

Reply:
Thanks @SuperSandro2000 !

Currently those modules are not built with nix.